# Module 1 Reflective Practice Assignment 

Spend between 20 and 30 minutes just experimenting with the python concepts that we have learned this week.  Get creative!  Try new things! 
Fail and try again!  Suggested practice items:

- Using mathematical operations to add, subtract, multiply, divide, or find the remainder 

- Using if statements with comparison operators such as >, >=, <, <=, ==, or!= 

Once you're done, save this Jupyter Notebook - you will submit it as part of this assignment. It is perfectly acceptable to describe trying to get something to work, and then failing to get it functioning as desired in the 20-30 min time block.  This does not mean you'll get a bad grade for this assignment -- failing is a necessary part of learning, though none of us particularly enjoy it.  Sometimes what you'll try to accomplish is just too big of a task for a Reflective Practice session, and that is ok -- you will still have learned from the experience.

## You can experiment with your code in the notebook cells below.

In [1]:
# add to variables together
num1 = 3.0
num2 = 3.4
add = num1 + num2 

print(add)

6.4


In [2]:
# Subtraction 
sub = num1 - num2
print(sub)

# Error in the result when first run 
# Use formatted string
print (f'{sub :.2f}')

-0.3999999999999999
-0.40


In [3]:
# divide  using integer and then float
num3 = 12
num4 = 24

# integer division (floor division))
div1 = num3 // num4

print(div1)

# Floating division
div2 = num3 / num4
print (div2)

0
0.5


In [6]:
# modulo arithmetic
# Find even or odd numbers
mod_num = int(input('Please enter an integer: '))
mod_num2 = int(input('Please enter a second number: '))

if mod_num % mod_num2 > 0:
    print("Not evenly divisible!")
if mod_num % mod_num2 == 0:
    print("Evenly divisible!")
                


Please enter an integer:  23
Please enter a second number:  21


Not evenly divisible!


#### Please place all code experimentation above this comment. ####

# Module 1 Reflection Submission

Reflect on what you have learned below.  Write a short (4-8 sentence summary) reflecting on what you figured out through this practice session. 

#### Enter Your Reflection Under this Cell ####

During this assignment there were a few interesting items that showed up. While printing the result for subtraction it was expected to get .40, but this was not the case. It required the use of a f-string to accomplish the correct output. While creating the if cases I erroneously entered the wrong comparison statement for the desired effect.  I entered ">=" instead of ">".  The way these conditionals were formatted if the number were evenly divisible then both of conditionals would be executed.  This would yield the wrong result or output.

#### Enter Your Reflection Above this Cell ####